src/tools/compiletest/src/runtest.rs

```diff
@@ -1587,13 +1588,15 @@ impl<'test> TestCx<'test> {
             passes,
         );
 
-        self.compose_and_run_compiler(rustc, None)
+        self.compose_and_run_compiler(rustc, None, self.testpaths)
     }
 
-    fn document(&self, out_dir: &Path) -> ProcRes {
+    /// `root_out_dir` and `root_testpaths` refer to the parameters of the actual test being run.
+    /// Auxiliaries, no matter how deep, have the same root_out_dir and root_testpaths.
+    fn document(&self, root_out_dir: &Path, root_testpaths: &TestPaths) -> ProcRes {
```

Comment:
Question: do rustdoc tests support revisions like ui tests, if so, would this be a problem with revisioned rustdoc tests?

Reply:
> I’ve never seen them used, and htmldocck itself doesn’t know about them, so I guess not.

Actually it's very easy:

https://github.com/rust-lang/rust/blob/1ddedbaa5919b7b3e70d984660e21e844c615c97/src/tools/compiletest/src/runtest.rs#L2661-L2662

no revisions are not supported for `tests/rustdoc` 😄